src/monster.cpp

```diff
@@ -761,8 +762,9 @@ bool monster::can_act() const
 int monster::sight_range( const int light_level ) const
 {
     // Non-aquatic monsters can't see much when submerged
-    if( !can_see() || ( underwater && !has_flag( MF_SWIMS ) && !has_flag( MF_AQUATIC ) &&
-                        !digging() ) ) {
+    if( !can_see() || has_effect( effect_no_sight ) || ( underwater && !has_flag( MF_SWIMS ) &&
+            !has_flag( MF_AQUATIC ) &&
+            !digging() ) ) {
```

Comment:
The line breaks here are not sensibly placed.

Reply:
I used the astyle extension on Visual studio and the settings from code_style.md
But I can re arrange things if needed

Reply:
What would be sensibles line break ? Keeping `!has_flag( MF_AQUATIC )` and `!digging()` on the same line ?

Reply:
I'd suggest either
```
if( !can_see() || has_effect( effect_no_sight ) ||
    ( underwater && !has_flag( MF_SWIMS ) && !has_flag( MF_AQUATIC ) && !digging() ) ) {
```
or
```
bool swimming = has_flag( MF_SWIMS ) || has_flag( MF_AQUATIC );
if( !can_see() || has_effect( effect_no_sight ) ||
    ( underwater && !swimming && !digging() ) ) {
```

Reply:
Done. I used the first suggestion.

Reply:
But maybe I should have edited `can_see()` directly ?